code/src/reducers/game.js

```diff
@@ -0,0 +1,79 @@
+import { createSlice } from "@reduxjs/toolkit"
+import { ui } from "./ui";
+
+export const game = createSlice({
+    name: 'game',
+    initialState: {
+      username: '',
+       gameObject: {
+         coordinates: '',
+         description: '',
+         actions: [],
+        },     
+      history:[]
+    },
+    reducers: {
+      setGameObject: (state, action) => {
+        if(state.gameObject) {
+          state.history.push(state.gameObject)
+        }
+        state.gameObject = action.payload;
+      },
+
+      setUserName: (state, action) => {
+        state.username = action.payload;
+      },
+    },
+  });
+
+
+  export const fetchGame = () => {
+    return (dispatch, getState) => {
+  
+      const options = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ 
+          username: getState().game.username }),
+      };
+      dispatch(ui.actions.setLoading(true))
+      fetch('https://labyrinth-technigo.herokuapp.com/start', options)
+        .then((res) => res.json())
+        .then((data) => {
+          dispatch(game.actions.setGameObject(data));
+          console.log(data)
```

Comment:
You might want to consider removing the console logs :)

Reply:
Thank you for the reminder!